### PR TITLE
Added documentation for Extended Crafting

### PR DESCRIPTION
**Pull request by JuiceyBeans**
Note: Table crafting is not added yet

### CODE REVIEW

File: docs/groovyscript/mods/extendedcrafting/compressioncrafting.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    .inputCount(100) //Requires 100 clay blocks
+    .output(item('minecraft:gold_ingot') * 7) //Outputs 7 gold ingots
+    .catalyst(item('minecraft:diamond')) // Uses a diamond as catalyst
+    .consumeCatalyst(true) //Consumes the catalyst (diamond)
+    .powerCost(10000)//Requires 10,000 FE total to run
+    .powerRate(1000) //Requires 1,000 FE/t to run
+    .register() //Registers your recipe. Very important!
+```
+
+## Removing recipes
```
Comment (WaitingIdly) [inline docs/groovyscript/mods/extendedcrafting/compressioncrafting.md]: i would suggest using a format like the one used for the Blood Altar, where it states what it does and then has a codeblock of the function call

https://github.com/juraj-hrivnak/GroovyScript-Docs/blob/746254fe18bd693bbc710784489cfa0e4a7891f1/docs/groovyscript/mods/bloodmagic/blood_altar.md?plain=1#L91

**Review by WaitingIdly — CHANGES_REQUESTED** [review]
1. the wiki isnt just markdown, it uses a specialized format for mkdocs. youll need to test this using the instructions in the readme.
2. you have to add the file locations to the `mkdocs.yml`
3. file names should be `snake_case`
4. text you want in a `!!!` block must be indented
5. there should be an empty line after each title/subtitle
6. include the package name for the packages, eg my example below
7. instead of an `### Example`, use an `!!! example`. 


````
Package:

```groovy
mods.extendedcrafting.Combination
mods.extendedcrafting.CombinationCrafting
```
````

File: docs/groovyscript/mods/extendedcrafting/combination_crafting.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+mods.extendedcrafting.CombinationCrafting
+``` 
+
+## Adding recipes
+Works only for items and energy
+
+```groovy
+mods.extendedcrafting.combination.recipeBuilder(ItemStack output, long cost, Ingredient input, NonNullList<Ingredient> pedestals)
+
+mods.extendedcrafting.combination.recipeBuilder(ItemStack output, long cost, int perTick, Ingredient input, NonNullList<Ingredient> pedestals)
```
Comment (brachy84) [inline docs/groovyscript/mods/extendedcrafting/combination_crafting.md]: `add` not `recipebuilder`

File: docs/groovyscript/mods/extendedcrafting/combination_crafting.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+This removes ALL recipes that match the given output:
+
+```groovy
+mods.bloodmagic.CombinationCrafting.removeByOutput(ItemStack)
+```
+
+Removes all registed Combination Crafting recipes:
+
+```groovy
+mods.bloodmagic.CombinationCrafting.removeAll()
```
Comment (brachy84) [inline docs/groovyscript/mods/extendedcrafting/combination_crafting.md]: bloodmagic?

File: docs/groovyscript/mods/extendedcrafting/compression_crafting.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+``` 
+
+## Adding recipes
+
+Works only for items and energy
+
+```groovy
+mods.extendedcrafting.combination.recipeBuilder(ItemStack output, IIngredient input, int inputCount, IIngredient catalyst, boolean consumeCatalyst, int powerCost)
+
+mods.extendedcrafting.compression.recipeBuilder(ItemStack output, IIngredient input, int inputCount, IIngredient catalyst, boolean consumeCatalyst, int powerCost, int powerRate)
```
Comment (brachy84) [inline docs/groovyscript/mods/extendedcrafting/compression_crafting.md]: combination? recipeBuilder?

**Review by brachy84 — CHANGES_REQUESTED** [review]

File: docs/groovyscript/mods/extendedcrafting/combination_crafting.md (hunk 3 above)
Comment (WaitingIdly) [reply docs/groovyscript/mods/extendedcrafting/combination_crafting.md]: oh whoops that ones on me

**Comment by brachy84** [comment]
superseeded by #18

**Pull request closed without merging**